systemtests/src/main/java/io/enmasse/systemtest/platform/apps/SystemtestsKubernetesApps.java

```diff
@@ -919,6 +918,7 @@ private static Deployment getMessagingAppDeploymentResource() {
                 .withImage("quay.io/enmasse/systemtests-clients:latest")
                 .withCommand("sleep")
                 .withArgs("infinity")
+                .withEnv(new EnvVarBuilder().withName("PN_TRACE_FRM").withValue("true").build())
```

Comment:
It makes me nervous the build on the protocol trace for this feature.  My main concerns:

1. It is not the protocol trace's intended purpose
1. AMQP uses links for other purposes (such as transaction coordination http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html), you run the risk of set the Future prematurely owing to the wrong attach being seen.
1. The protocol trace format may change.

That said, may be this is pragmatic step towards a solution we want.  We could always refactor the messaging clients later, one by one, adding 'proper' support then switch it out in the client implementations.







 AMQP uses links for other features - for instance, transaction in AMQP are coordinated via the transaction coordinator

Reply:
Agree on that, atm this is the best approach we can have given with the functionalities the clients currently have